ip-utils.h

```diff
@@ -0,0 +1,139 @@
+/*! \file    ip-utils.h
+ * \author   Johan Ouwerkerk <jm.ouwerkerk@gmail.com>
+ * \copyright GNU General Public License v3
+ * \brief    IP address related utility functions (headers)
+ * \details  Provides functions to query for network devices with a given device name or address.
+ * \p Devices may be looked up by either a device name or by the IPv4 or IPv6 address of the configured network interface.
+ * This functionality may be used to bind to user configurable network devices instead of relying on unpredictable implementation defined defaults.
+ *
+ * \p Parsing IPv4/IPv6 addresses is robust against malformed input.
+ *
+ * \see man 3 getifaddrs
+ * \see man 3 inet_pton
+ * \ingroup core
+ * \ref core
+ */
+
+#ifndef _JANUS_IP_UTILS_H
+#define _JANUS_IP_UTILS_H
+
+#include <ifaddrs.h>
+#include <netinet/in.h>
+
+/** @name Janus helper methods to match names and addresses with network interfaces/devices.
+ */
+///@{
+typedef enum JanusNetworkQueryOptions {
+	NoMatch = 0, NameOnlyMatch = 1, IPv4OnlyMatch = 2, IPv6OnlyMatch = 4, AnyIPMatch = 6,  AnyMatch = 7
```

Comment:
Same as above for enum names, which should at least in part include the enum type (e.g., `NoMatch` --> `janus_iputils_network_query_option_no_match`).